ground/src/main/java/com/google/android/ground/persistence/remote/firestore/GeometryConverter.kt

```diff
@@ -84,17 +86,29 @@ class GeometryConverter {
     /**
      * Converts a `Map` deserialized from Firestore into a `Geometry` instance.
      */
-    fun fromFirestoreMap(map: Map<String, Any>): Geometry? {
+    fun fromFirestoreMap(map: Map<String, *>?): Geometry? {
+        if (map == null) return null
         val jsonMap = fromFirestoreValue(map)
-        val jsonString = Gson().toJson(jsonMap)
-        val reader = GeoJsonReader()
-        return reader.read(jsonString)
+        try {
+            val jsonString = Gson().toJson(jsonMap)
+            val reader = GeoJsonReader()
+            val geometry = reader.read(jsonString)
+            if (geometry.coordinates.isEmpty()) {
+                throw DataStoreException("Empty coordinates in $geometry")
+            }
+            return geometry
+        } catch (e: Throwable) {
+            throw DataStoreException("Invalid geometry")
```

Comment:
Why not log this error and just return `null` in this case?

Reply:
A better option: If we think it's important to distinguish `null` values from "something was malformed" values, we should use [`Result`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/). Result will also leave the decision about whether or not a failure in this code is a truly exceptional condition up to callers (see `exceptionOrNull`)

Exceptions are not properly in the class of things that are values and correspond to non-terminating conditions (even though they are really unexpected terminations--in kotlin they are typed as `Nothing`, the same type as an infinite loop, the type of all "unreachable" code: https://kotlinlang.org/docs/exceptions.html#the-nothing-type).

imo, in Kotlin code, we should be a bit more conservative about throwing and only do so when we really want to reflect that some program state effectively results in improper operation of the whole system. In this case I'd expect the app to continue to function just fine if one ore two geometries are malformed.

Reply:
Applying suggestion to use `Result` instead